eng/pipelines/rolling-internal-validation-pipeline.yml

```diff
@@ -61,18 +73,53 @@ extends:
       tsa:
         enabled: ${{ not(parameters.disableTSA) }}
         configFile: $(Build.SourcesDirectory)/.config/tsa/tsaoptions.json
+      sourceRepositoriesToScan:
+        include:
+          - repository: GoMirror
 
     stages:
-      - stage: Analyze
+      # Don't run CodeQL if this is triggered by a push to GoMirror.
+      - ${{ if in(variables['Build.Reason'], 'Schedule', 'Manual') }}:
+        - stage: GoInfraCodeQL
+          displayName: go-infra CodeQL
+          dependsOn: []
+          jobs:
+            - job: Analyze
+              workspace:
+                clean: all
+              steps:
+                - checkout: self
+
+                - task: GoTool@0
+                  inputs:
+                    version: 1.24.1
+
+                - script: |
+                    go build ./...
+                  displayName: go
+
+      - stage: MirrorCG
+        displayName: go-mirror CG
+        dependsOn: []
         jobs:
-          - job: Analyze
+          - job: MirrorCG
             workspace:
               clean: all
+            variables:
+              - name: RepositoryUrl
+                value: $[resources.repositories.GoMirror.url]
+              - name: RepositoryVersion
+                value: $[resources.repositories.GoMirror.version]
+              # We check out the repository ourselves, so we need to manually
+              # specify the repository information.
+              - name: CG_ADO_BUILD_REPOSITORY_URI
+                value: $[resources.repositories.GoMirror.url]
+              - name: CG_ADO_BUILD_SOURCEBRANCH
+                value: $[resources.repositories.GoMirror.ref]
+              - name: CG_ADO_BUILD_SOURCEVERSION
+                value: $[resources.repositories.GoMirror.version]
             steps:
-              - task: GoTool@0
-                inputs:
-                  version: 1.22.1
-
-              - script: |
-                  go build ./...
-                displayName: go
+              - template: /eng/pipelines/steps/checkout-windows-task.yml@self
```

Comment:
I believe using `checkout: GoMirror` hits https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/secure-supply-chain/secure-supply-chain-analysis-task/secure-supply-chain-analysis because there are non-policy files that exist in upstream that we can't remove. A custom checkout seems to avoid this.